docs/classification/calibration.md

```diff
@@ -0,0 +1,117 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Probability Calibration
+
+A classification task predicts a label for a given observation based on some historical data. An alternative to predicting the class directly is to predict the probability of an observation belonging to each possible class.
+
+Calibration curves (also known as reliability diagrams), plot the true frequency of the positive label against its predicted probability, for binned predictions. Specifically, the predicted probabilities are divided up into a fixed number of buckets along the x-axis. The number of events of positive class are counted for each bin and then these counts are normalised.
+
+The better calibrated or more reliable a forecast, the closer the points will appear along the main diagonal from the bottom left to the top right of the plot.
+
+Calibration of prediction probabilities is a rescaling operation that is applied after the predictions have been made by a predictive model.
+
+Calibration of a classifier can be checked by creating a reliability diagram as demonstrated below.
+
++++
+
+## Reliability Diagrams
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+```
+
+```{code-cell} ipython3
+from sklearn.datasets import make_classification
+from sklearn.model_selection import train_test_split
+
+X, y = make_classification(
+    n_samples=20000, n_features=4, n_informative=2, n_redundant=2, class_sep=0.7, random_state=0)
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, test_size=0.33, random_state=0
+)
+```
+
+```{code-cell} ipython3
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.svm import SVC
+
+svc = make_pipeline(StandardScaler(), SVC(gamma='auto', probability=True))
+svc_probas = svc.fit(X_train, y_train).predict_proba(X_test)
+```
+
+```{code-cell} ipython3
```

Comment:
organize this in 2-3 cells and add some explanation before each cell, otherwise it's too many steps to copy-paste each cell